src/common/utils/utils.ts

```diff
@@ -52,7 +52,11 @@ function getLastMidnightTimestamps(daysBack: number, startFromTonight = false):
 
 // TODO: replace these functions with internationalization functions
 // always round USD amounts to two decimals
-function getUsdAmount<C extends CurrencyUnit>(amount: MonetaryAmount<Currency<C>, C>, rate: number): string {
+function getUsdAmount<C extends CurrencyUnit>(amount: MonetaryAmount<Currency<C>, C>, rate: number | undefined): string {
```

Comment:
This might be related to some preference. But if you ask me, I'd rather type `rate` like this after renaming `rate` `price`:
```
price: { usd: number } | undefined
```
To be honest, personally I don't like using the optional chaining operator much since it might help us avoid unexpected cases like null or undefined but it's implicit like JavaScript vs. TypeScript.

If I could pick up one more thing, it might be good for maintenance since we can adapt to E.g. price: { gbp: number; } which is unlikely to happen to be honest though.

Reply:
Thanks for the review @crypto-engineer! I think the current solution is good enough and if we decide to add another currency to display prices in, we can always change it in the future :)